### PR TITLE
Update global_4denvar and global_enkf ctests to reflect GFS v17

### LINKED ISSUE #759: Update global_4denvar and global_enkf ctests to reflect GFS v17
The global_4denvar (`gsi.x`) and global_enkf (`enkf.x`) scripts and namelists 
- `regression/global_4denvar.sh`
- `regression/global_enkf.sh`
- `regression/regression_namelists.sh`
- `regression/regression_namelists_db.sh`

should reflect how we intend to run `gsi.x` and `enkf.x` in GFS v17.   This issue is opened to document these changes.

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
**Description**
Update global_4denvar and global_enkf ctest configurations to better align with GFS v17 configuration.

Resolves #759 

**Type of change**
- [x] Maintenance - update global ctest configuration

**How Has This Been Tested?**
1. install RussTreadon-NOAA/GSI:feature/global_ctests on Hera
2. run global_4envar and global_enkf ctests

Both tests pass
```
Test project /scratch1/NCEPDEV/da/Russ.Treadon/git/gsi/global_ctests/build
    Start 1: global_4denvar
1/2 Test #1: global_4denvar ...................   Passed  1926.51 sec
    Start 6: global_enkf
2/2 Test #6: global_enkf ......................   Passed  755.75 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) = 2682.29 sec
```
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] New and existing tests pass with my changes

### CODE REVIEW

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 (Cactus) test**
Install `RussTreadon-NOAA:feature/global_ctests` on Cactus.  Run global  ctests with following results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/pr774/build
    Start 1: global_4denvar
1/2 Test #1: global_4denvar ...................   Passed  1684.42 sec
    Start 6: global_enkf
2/2 Test #6: global_enkf ......................   Passed  852.93 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) = 2537.53 sec
```

**Comment by RussTreadon-NOAA** [comment]
**Hercules test**
Install `RussTreadon-NOAA:feature/global_ctests` on Hercules. Run global ctests with following results
```
Test project /work/noaa/da/rtreadon/git/gsi/pr774/build
    Start 1: global_4denvar
1/2 Test #1: global_4denvar ...................   Passed  2222.24 sec
    Start 6: global_enkf
2/2 Test #6: global_enkf ......................   Passed  1206.22 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) = 3428.47 sec
```

**Review by azadeh-gh — APPROVED** [review]
looks good.

File: regression/global_enkf.sh [hunk 1]
```diff
@@ -76,14 +76,14 @@ export use_correlated_oberrs=${use_correlated_oberrs:-".false."}
 if [ $USE_CORRELATED_OBERRS == "YES" ]; then
     export use_correlated_oberrs=".true."
 fi
-export imp_physics=${imp_physics:-"11"}
+export imp_physics=${imp_physics:-"8"}
 export lupp=${lupp:-".true."}
 export corrlength=${corrlength:-1250}
 export lnsigcutoff=${lnsigcutoff:-2.75}
 export analpertwt=${analpertwt:-0.85}
 export readin_localization_enkf=".false."
```
Comment (CatherineThomas-NOAA) [inline regression/global_enkf.sh:83]: Do you know why this line was here separately rather than changing the default?  Is it still needed?

Comment (CatherineThomas-NOAA) [reply regression/global_enkf.sh:83]: The line I'm referring to is Line 84: export readin_localization_enkf=".false."

**Comment by CatherineThomas-NOAA** [comment]
@RussTreadon-NOAA:  Should the debug test also be updated?  I see you added the imp_physics change, but it looks like that test is quite out of sync with regression_namelist.sh.  I have the script comparison on Hera here:
`/scratch1/NCEPDEV/da/Catherine.Thomas/gsi/russ_pr/regression/diff`

Different parameters between scripts:
- iterations
- RO parameters: nsig_ext, gpstop, commgps*
- incvars*
- obslist

Missing parameters in regression_namelist_db.sh: 
- optconv=0.05,cao_check=.true.,ta2tb=.true.,
- tcp*, *cads 
- dmesh(4)
- hub_norm

File: regression/global_enkf.sh (hunk 1 above)
Comment (RussTreadon-NOAA) [reply regression/global_enkf.sh:83]: I assume the namelist variables in this script trace back to when this script was used as a stand-alone script to run `enkf.x`.  This script is now used for ctests.  Remove the following lines from `global_enkf.sh`
```
-export imp_physics=${imp_physics:-"8"}
-export readin_localization_enkf=${readin_localization_enkf:-".false."}
-export reducedgrid=${reducedgrid:-".false."}
```
Also update global_enkf section of `regression_namelists.sh` to explicitly set `imp_physics=8` as is done for global_4denvar.

Done at [`11a80d6`](https://github.com/RussTreadon-NOAA/GSI/commit/11a80d62935fac29baeb1248a4c58efd0dc7287b).

**Comment by RussTreadon-NOAA** [comment]
@CatherineThomas-NOAA , thank you for noting differences between the namelists in `regression_namelists.sh` and `regression_namelists_db.sh`.  

The debug option traces back to svn.  I am not aware of anyone who uses the debug ctests.  I don't.  When I want to test changes in debug mode, I build with `BUILD_TYPE=Debug` and run the standard suite of ctests.  The wall times need to be increased when doing this.  I'd be fine with removing the `debug` flag and associated scripting (namelists) from `regression/`. 

The GSI Review team has talked about an overhaul of the GSI ctest suite.  The checks performed by the test scripts need to be refactored.  The current checks generate too many false positives. Run scripts should be cleaned up and standardized.  Thus far, no one has found time to go through this cleanup, enhancing, and refactoring task.

**Comment by RussTreadon-NOAA** [comment]
Build `RussTreadon-NOAA:feature/global_ctests` at 11a80d6 on Cactus.  Run `global_` ctests with the folowing results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/global_ctests/build
    Start 1: global_4denvar
1/2 Test #1: global_4denvar ...................   Passed  1683.88 sec
    Start 6: global_enkf
2/2 Test #6: global_enkf ......................   Passed  853.32 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) = 2537.24 sec
```

**Review by CatherineThomas-NOAA — APPROVED** [review]
Thanks for your comments @RussTreadon-NOAA.  Being able to run in debug mode using the standard ctests makes much more sense than using a separate debug script where we have to make changes in multiple places.  Looks good to me.  Approve.

**Pull request merged**